experimental/aitools/server.go

```diff
@@ -0,0 +1,254 @@
+package aitools
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/databricks/cli/experimental/aitools/tools"
+	"github.com/databricks/cli/libs/log"
+	"github.com/spf13/cobra"
+)
+
+func newServerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "server",
+		Short: "Start the MCP server (used by coding agents)",
+		Long:  `Start the Databricks CLI MCP server. This command is typically invoked by coding agents.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runServer(cmd.Context())
+		},
+	}
+
+	return cmd
+}
+
+type jsonrpcRequest struct {
+	JSONRPC string          `json:"jsonrpc"`
+	ID      any             `json:"id,omitempty"`
+	Method  string          `json:"method"`
+	Params  json.RawMessage `json:"params,omitempty"`
+}
+
+type jsonrpcResponse struct {
+	JSONRPC string    `json:"jsonrpc"`
+	ID      any       `json:"id,omitempty"`
+	Result  any       `json:"result,omitempty"`
+	Error   *rpcError `json:"error,omitempty"`
+}
+
+type rpcError struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+	Data    any    `json:"data,omitempty"`
+}
+
+// JSON-RPC 2.0 error codes.
+const (
+	jsonRPCParseError     = -32700
+	jsonRPCInvalidRequest = -32600
+	jsonRPCMethodNotFound = -32601
+	jsonRPCInvalidParams  = -32602
+	jsonRPCInternalError  = -32603
+)
+
+type mcpServer struct {
```

Comment:
Are you implementing the protocol manually?

If so it would be helpful to extract this into a separate file, which I can copy into my version.